src/frontend/sharedComponents/SelectDevice.tsx

```diff
@@ -0,0 +1,226 @@
+import React from 'react';
+import {defineMessages, useIntl} from 'react-intl';
+import {ScrollView, StyleSheet, View, TouchableOpacity} from 'react-native';
+
+import {useManyMembers} from '@comapeo/core-react';
+import {type MemberInfo} from '@comapeo/core/dist/member-api';
+import {type MapeoClientApi} from '@comapeo/ipc';
+import {useLocalDiscoveryState} from '../hooks/useLocalDiscoveryState';
+import {useLocalPeers} from '../hooks/useLocalPeers';
+import WifiIcon from '../images/WifiIcon.svg';
+import {DeviceNameWithIcon} from './DeviceNameWithIcon';
+import {BodyText} from './Text/BodyText';
+import {HeaderText} from './Text/HeaderText';
+import {useActiveProject} from '../contexts/ActiveProjectContext';
+import {LIGHT_GREY} from '../lib/styles';
+import {ExhaustivenessError} from '../lib/ExhaustivenessError';
+import {type NativeRootNavigationProps} from '../sharedTypes/navigation';
+
+type PublicPeerInfo = Awaited<
+  ReturnType<MapeoClientApi['listLocalPeers']>
+>[number];
+
+const m = defineMessages({
+  title: {
+    id: 'screen.Settings.ProjectSettings.YourTeam.SelectDevice.title',
+    defaultMessage: 'Select Device to Invite',
+  },
+  titleMapShare: {
+    id: 'screen.Settings.ProjectSettings.YourTeam.SelectDevice.titleMapShare',
+    defaultMessage: 'Select Device',
+  },
+  notSeeingDevice: {
+    id: 'screen.Settings.ProjectSettings.YourTeam.SelectDevice.notSeeingDevice',
+    defaultMessage: 'Not seeing a Device?',
+  },
+  sameWifi: {
+    id: 'screen.Settings.ProjectSettings.YourTeam.SelectDevice.sameWifi',
+    defaultMessage: 'Make sure both devices are on the same wifi network ',
+  },
+  sameVersion: {
+    id: 'screen.Settings.ProjectSettings.YourTeam.SelectDevice.sameVersion',
+    defaultMessage: 'Make sure both devices are on the same version of CoMapeo',
+  },
+});
+
+type SelectionMode = 'invites' | 'shareMap';
+
+export const SelectDevice = ({
+  navigation,
+  route,
+}: NativeRootNavigationProps<'SelectDevice' | 'SelectMapShareDevice'>) => {
+  const ssid = useLocalDiscoveryState(state => state.ssid);
+  const {formatMessage: t} = useIntl();
+
+  const devices = useInitiallyConnectedPeers();
```

Comment:
I know you didn't originally name this, but can we rename devices to "availablePeers"